src/frontend/src/catalog/schema_catalog.rs

```diff
@@ -160,7 +160,11 @@ impl SchemaCatalog {
     }
 
     pub fn drop_table(&mut self, id: TableId) {
-        let table_ref = self.table_by_id.remove(&id).unwrap();
+        let Some(table_ref) = self.table_by_id.remove(&id) else {
+            tracing::warn!(?id, "table to drop not found, cleaning up?");
```

Comment:
Why do we loosen the check to just a `warning` instead?

Reply:
See https://github.com/risingwavelabs/risingwave/pull/18944#discussion_r1808055899